ably/realtime_conn.go

```diff
@@ -50,12 +65,13 @@ type connCallbacks struct {
 
 func newConn(opts *ClientOptions, auth *Auth, callbacks connCallbacks) (*Connection, error) {
 	c := &Connection{
-		opts:      opts,
-		msgCh:     make(chan *proto.ProtocolMessage),
-		state:     newStateEmitter(StateConn, StateConnInitialized, "", auth.logger()),
-		pending:   newPendingEmitter(auth.logger()),
-		auth:      auth,
-		callbacks: callbacks,
+		opts:        opts,
+		recoveryKey: opts.Recover,
```

Comment:
Why create a new field when we can get it from `opts`?

Reply:
addressed in 4449023b755bb04b26b63f4dab7c83ccd5dba667